internal/controller/networking/ingress_controller.go

```diff
@@ -187,13 +219,156 @@ func (r *IngressReconciler) gatherApiCheckData(ingress *networkingv1.Ingress) (a
 		muted = true
 	}
 
-	apiCheckSpec = checklyv1alpha1.ApiCheckSpec{
-		Endpoint: endpoint,
-		Group:    group,
-		Success:  success,
-		Muted:    muted,
+	labels := make(map[string]string)
+	labels["ingress-controller"] = ingress.Name
+
+	// Get the host(s) and path(s) from the ingress object
+	// No Rules specified, nothing to do
+	if len(ingress.Spec.Rules) == 0 {
+		return
+	}
+
+	for _, rule := range ingress.Spec.Rules {
+
+		// Get the host
+		var host string
+		if ingress.Annotations[annotationEndpoint] == "" {
+			host = rule.Host
+		} else {
+			host = ingress.Annotations[annotationEndpoint]
+		}
```

Comment:
So this pattern seem to be quite prevalent in the codebase, if you prefer it the way it is then alright, but I would probably just do this instead:

```suggestion
		host := ingress.Annotations[annotationEndpoint]
		if host == "" {
			host = rule.Host
		}
```

Always good if you're able to reduce redundant map accesses because it gives you less opportunity to use the wrong key.

Reply:
Fixed in latest commit.